hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -277,7 +277,7 @@ public boolean dropNamespace(Namespace namespace) {
       return true;
 
     } catch (InvalidOperationException e) {
-      throw new NamespaceNotEmptyException("Namespace " + namespace + " is not empty. One or more tables exist.", e);
+      throw new NamespaceNotEmptyException(e, "Namespace %s is not empty. One or more tables exist.", namespace);
```

Comment:
Previous usage was broken

Reply:
Maybe we should use a factory method that checks whether the last argument is an exception and calls the constructor appropriately. Then we wouldn't have to remember to do this.

Reply:
The new checks will throw a build error if you do it wrong :)

Reply:
Great!

Reply:
Hooray for compile time safety! My inner scala developer is pleased. 😀